javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -333,11 +333,20 @@ public static CharSeq repeat(char character, int times) {
      * @return A CharSeq representing {@code this * times}
      */
     public CharSeq repeat(int times) {
-        final StringBuilder builder = new StringBuilder();
-        for (int i = 0; i < times; i++) {
-            builder.append(back);
+        if (times <= 0) return empty();
+        else if (times == 1) return this;
+        else {
+            final int finalLength = length() * times;
+            final StringBuilder result = new StringBuilder(2 * finalLength);
+            result.append(back);
+
+            for (int i = 1; i <= times; i *= 2) {
+                result.append(result);
+            }
+            result.setLength(finalLength);
```

Comment:
Nice solution! Guava does it directly with arrays instead of using a StringBuilder. Also Guava's size check and the last array copy (to save mem) are nice:

https://github.com/google/guava/blob/c462d69329709f72a17a64cb229d15e76e72199c/guava/src/com/google/common/base/Strings.java#L142-L166

What do you think, give it a try the 'Guava way'?